presto-main/src/main/java/com/facebook/presto/util/maps/IdentityLinkedHashMap.java

```diff
@@ -142,4 +133,116 @@ public int size()
             }
         };
     }
+
+    private class KeySet
```

Comment:
What prompted this change? When this class was first added I remember thinking it looked odd but it didn't seem broken, just limited/potentially inefficient (mutation methods in AbstractSet throw, everything else delegates to iterator()).